packages/skill-api/src/core/services/process-stripe-webhook.ts

```diff
@@ -406,12 +406,17 @@ export const processStripeWebhook = async (
         ? targetEmail.toLowerCase() !== previousEmail.toLowerCase()
         : false
 
+      const {user: fromUser} = previousEmail
+        ? await findOrCreateUser(previousEmail)
+        : {user: null}
       const {user: updateUser} = await findOrCreateUser(targetEmail, name)
 
-      await transferPurchasesToNewUser({
-        merchantCustomerId: merchantCustomer.id,
```

Comment:
this was the root cause because the incoming merchantCustomer row was shared across 20,557 charges and purchases for reasons

now there is no shared merchant customer and this is also more specific